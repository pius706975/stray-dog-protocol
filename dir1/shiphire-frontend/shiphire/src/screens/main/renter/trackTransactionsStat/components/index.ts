export { default as ShipInformationCard } from "./ShipInformationCard"
export { default as TransactionProgress } from "./TransactionProgres"