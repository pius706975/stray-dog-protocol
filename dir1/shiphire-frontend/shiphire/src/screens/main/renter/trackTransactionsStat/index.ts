export { default as TrackTransactionsStat } from "./TrackTransactionsStat"