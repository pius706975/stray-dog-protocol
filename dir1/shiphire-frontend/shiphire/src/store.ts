import { configureStore } from '@reduxjs/toolkit';
import {
    addShipSlice,
    documentNameSlice,
    modalSlice,
    networkErrorSlice,
    notifBadgeSlice,
    progressIndicatorSlice,
    renterPreferenceSlice,
    requestForQuoteSlice,
    shipSpecSlice,
    termsStatusSlice,
    userLanguageSlice,
    userLocationSlice,
    userStatusSlice,
} from './slices';

const store = configureStore({
    reducer: {
        modal: modalSlice.reducer,
        progressIndicator: progressIndicatorSlice.reducer,
        renterPreference: renterPreferenceSlice.reducer,
        requestForQuote: requestForQuoteSlice.reducer,
        userStatus: userStatusSlice.reducer,
        addShip: addShipSlice.reducer,
        documentName: documentNameSlice.reducer,
        termsStatus: termsStatusSlice.reducer,
        userLanguage: userLanguageSlice.reducer,
        userLocation: userLocationSlice.reducer,
        shipSpecSlice: shipSpecSlice.reducer,
        notifBadgeSlice: notifBadgeSlice.reducer,
        networkError: networkErrorSlice.reducer,
    },
});

export default store;
