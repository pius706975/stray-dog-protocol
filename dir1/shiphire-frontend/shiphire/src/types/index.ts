export * from './Components'
export * from './Navigators'
export * from './ReduxToolkit'
export * from './Screens'
export * from './Services'