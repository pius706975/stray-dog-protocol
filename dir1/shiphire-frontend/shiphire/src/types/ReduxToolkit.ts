import { DocumentPickerResponse } from 'react-native-document-picker';
import store from '../store';

export interface ModalState {
    visible: boolean;
    status: 'success' | 'failed' | 'info' | undefined;
    text: string;
}

export interface ProgressIndicatorState {
    visible: boolean;
}

export interface RenterPreferenceSlice {
    renterPreference: string[];
}

export interface RequestForQuoteSlice {
    requestForQuoteData: any[];
}

export interface AddShipSlice {
    name: string;
    desc: string;
    category: string;
    pricePerMonth: string;
    location: string;
    length: string;
    width: string;
    height: string;
    facilities?: (string | undefined)[];
    specifications?: { name: string; value: string }[];
    shipDocument: {
        uri: string;
        name: string | null;
        copyError?: string | undefined;
        fileCopyUri: string | null;
        type: string | null;
        size: number | null;
        label: string;
        docExpired?: string | undefined;
    }[];
}

export interface UserStatusState {
    isLoggedIn: boolean;
    isRoleSubmitted: boolean;
    isPreferencesSubmitted: boolean;
    isOwner: boolean;
    isCompanySubmitted: boolean;
    isAdmin: boolean;
    isCompanyVerif: boolean;
    isCompanyReject: boolean;
    isGoogleSignIn: boolean;
}

export interface DocumentNameState {
    text: string;
}

export interface TermsStatusState {
    status: boolean;
}

export interface UserLanguageState {
    language: string;
}

export interface IUserLocation {
    city: string;
    province: string;
}

export interface IUserCoordinates {
    latitude: string;
    longitude: string;
}

export interface UserLocationState {
    userLocation: IUserLocation;
    userCoordinates: IUserCoordinates;
}

export interface ShipSpecStateSlice {
    shipSpecSlice: string[];
}

export interface NotifBadgeSlice {
    totalNotif: number;
}

export type RootState = ReturnType<typeof store.getState>;
export type AppDispatch = typeof store.dispatch;

export interface NetworkErrorState {
    isNetworkError: boolean;
}