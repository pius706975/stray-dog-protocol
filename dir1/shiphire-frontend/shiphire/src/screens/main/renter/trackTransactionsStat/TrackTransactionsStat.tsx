import { useIsFocused } from '@react-navigation/native';
import React from 'react';
import { ScreenLayout } from '../../../../components';
import { useGetTransactionByRentalId } from '../../../../hooks';
import { TrackTrasactionsStatProps, Transaction } from '../../../../types';
import {
    getStatusText,
    getStatusTextDesc,
    handleAxiosError,
} from '../../../../utils';
import { ShipInformationCard, TransactionProgress } from './components';
import { useTranslation } from 'react-i18next';

const TrackTransactionsStat: React.FC<TrackTrasactionsStatProps> = ({
    route,
}) => {
    const { rentalId } = route.params;
    const isFocused = useIsFocused();
    const mutationGetTransactionById = useGetTransactionByRentalId();
    const [transaction, setTransaction] = React.useState<Transaction>();
    const formattedShipSize = `${transaction?.ship.size.length} m x ${transaction?.ship.size.width} m x ${transaction?.ship.size.height} m`;
    const status = new Array();
    const statusDate = new Array();
    const { t } = useTranslation('common');

    transaction?.status.map(item => {
        const statusClassification = item?.name?.split(' ')[0];
        if (
            item.name !== 'proposal 2' &&
            item.name !== 'contract 2' &&
            item.name !== 'complete' &&
            statusClassification !== 'payment' &&
            statusClassification !== 'sailing'
        ) {
            status.push(getStatusTextDesc(item.desc, t));
            statusDate.push(item.date);
        }

        status.push(getStatusText(item.name, t));
        statusDate.push(item.date);
    });

    React.useEffect(() => {
        isFocused &&
            mutationGetTransactionById.mutate(rentalId, {
                onSuccess: resp => {
                    setTransaction(resp.data.data);
                },
                onError: err => {
                    handleAxiosError(err);
                },
            });
    }, [isFocused]);

    return (
        <ScreenLayout
            backgroundColor="light"
            testId="TrackTransactionsStatScreen"
            padding={10}
            gap={10}>
            <ShipInformationCard
                formattedShipSize={formattedShipSize}
                shipImageUrl={transaction ? transaction.ship.imageUrl : ''}
                shipName={transaction?.ship.name}
                shipCategory={transaction?.ship.category.name}
            />
            <TransactionProgress
                rentalId={rentalId}
                status={status.reverse()}
                statusDate={statusDate.reverse()}
                statusLength={status.length}
            />
        </ScreenLayout>
    );
};

export default TrackTransactionsStat;
