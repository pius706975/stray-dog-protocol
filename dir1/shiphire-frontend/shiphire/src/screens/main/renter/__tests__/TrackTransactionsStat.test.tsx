import { createNativeStackNavigator } from '@react-navigation/native-stack';
import { QueryClient, QueryClientProvider } from 'react-query';
import { MainStackParamList } from '../../../../types';
import httpRequest from '../../../../services/api';
import MockAdapter from 'axios-mock-adapter';
import store from '../../../../store';
import { Provider } from 'react-redux';
import { NavigationContainer } from '@react-navigation/native';
import { TrackTransactionsStat } from '../trackTransactionsStat';
import { render } from '@testing-library/react-native';

const queryClient = new QueryClient();
const MainStack = createNativeStackNavigator<MainStackParamList>();
const mockAdapter = new MockAdapter(httpRequest);

const mockTransaction = {
    status: 'success',
    data: {
        ship: {
            size: {
                length: 12,
                width: 12,
                height: 12,
            },
            _id: 'shipId',
            shipOwnerId: 'shipOwnerId',
            name: 'shipName',
            category: {
                _id: 'categoryId',
                name: 'categoryName',
            },
            imageUrl: 'imageUrl',
            shipDocuments: [
                {
                    _id: 'documentId',
                    documentUrl: 'documentUrl',
                },
            ],
        },
        rfq: {
            rfqId: 'rfqId',
            rfqUrl: 'rfqUrl',
        },
        proposal: {
            proposalId: [],
            _id: 'proposalId',
            proposalUrl: 'proposalUrl',
            notes: 'notes',
            proposalName: 'proposalName',
            additionalImage: [],
        },
        contract: {
            contractId: 'contractId',
            contractUrl: 'contractUrl',
        },
        _id: 'transactionId',
        rentalId: 'rentalId',
        renterId: 'renterId',
        rentalDuration: 60,
        rentalStartDate: '2024-01-27T08:49:43.158+00:00',
        rentalEndDate: '2024-03-27T08:49:43.158+00:00',
        offeredPrice: 12,
        status: [
            {
                name: 'rfq',
                desc: 'Request for Quote',
                date: '2024-01-27T08:49:43.158+00:00',
                isOpened: true,
                _id: 'id',
            },
        ],
        createdAt: 'createdAt',
        updatedAt: 'updatedAt',
        ownerDetails: {
            company: {
                name: 'companyName',
            },
        },
        sailingStatus: 'sailingStatus',
        payment: [],
        beforeSailingPictures: [],
        afterSailingPictures: [],
        __v: 0,
    },
};

const trackStackNavMockComponent = () => (
    <QueryClientProvider client={queryClient}>
        <Provider store={store}>
            <NavigationContainer>
                <MainStack.Navigator>
                    <MainStack.Screen
                        name="TrackTransactionsStat"
                        component={TrackTransactionsStat}
                        initialParams={{
                            rentalId: 'rentalId',
                        }}
                    />
                </MainStack.Navigator>
            </NavigationContainer>
        </Provider>
    </QueryClientProvider>
);

mockAdapter
    .onGet('/renter/get-transaction-by-id/rentalId')
    .reply(200, mockTransaction);

describe('TrackTransactionsStat', () => {
    describe('snapshot testing', () => {
        it('should match snapshot', () => {
            const tree = render(trackStackNavMockComponent()).toJSON();
            expect(tree).toMatchSnapshot();
        });
    });
    describe('render screen correctly', () => {
        it('should render screen correctly', async () => {
            const { findByTestId } = render(trackStackNavMockComponent());
            const screen = await findByTestId('TrackTransactionsStatScreen');
            expect(screen).toBeTruthy();
        });
    });
});
