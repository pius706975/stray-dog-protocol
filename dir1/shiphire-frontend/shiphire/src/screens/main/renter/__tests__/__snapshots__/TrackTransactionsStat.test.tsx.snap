// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TrackTransactionsStat snapshot testing should match snapshot 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    {
      "flex": 1,
    }
  }
>
  <RNSScreenStack
    style={
      {
        "flex": 1,
      }
    }
  >
    <RNSScreen
      collapsable={false}
      forwardedRef={[Function]}
      gestureResponseDistance={
        {
          "bottom": -1,
          "end": -1,
          "start": -1,
          "top": -1,
        }
      }
      nativeBackButtonDismissalEnabled={false}
      onAppear={[Function]}
      onDisappear={[Function]}
      onDismissed={[Function]}
      onGestureCancel={[Function]}
      onHeaderBackButtonClicked={[Function]}
      onNativeDismissCancelled={[Function]}
      onTransitionProgress={[Function]}
      onWillDisappear={[Function]}
      replaceAnimation="push"
      sheetAllowedDetents="large"
      sheetCornerRadius={-1}
      sheetExpandsWhenScrolledToEdge={true}
      sheetGrabberVisible={false}
      sheetLargestUndimmedDetent="all"
      stackPresentation="push"
      style={
        {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
      swipeDirection="horizontal"
    >
      <View
        accessibilityElementsHidden={false}
        importantForAccessibility="auto"
        style={
          {
            "flex": 1,
            "flexDirection": "column-reverse",
          }
        }
      >
        <View
          collapsable={false}
          style={
            [
              {
                "flex": 1,
              },
              {
                "backgroundColor": "rgb(242, 242, 242)",
              },
              undefined,
            ]
          }
        >
          <View
            style={
              [
                undefined,
                undefined,
                {
                  "flex": 1,
                },
                undefined,
                {},
                {},
                {},
                {
                  "backgroundColor": "#f0f6fb",
                  "justifyContent": "space-between",
                  "position": "relative",
                },
              ]
            }
            testID="TrackTransactionsStatScreen"
          >
            <RCTScrollView
              contentContainerStyle={
                {
                  "alignItems": undefined,
                  "flex": undefined,
                  "flexGrow": 1,
                  "gap": 10,
                  "justifyContent": undefined,
                  "marginBottom": undefined,
                  "padding": 10,
                  "paddingVertical": undefined,
                }
              }
            >
              <View>
                <View
                  style={
                    [
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                      {
                        "backgroundColor": "#216dab",
                        "borderRadius": 6,
                        "padding": 16,
                      },
                    ]
                  }
                >
                  <View
                    row={true}
                    spread={true}
                    style={
                      [
                        undefined,
                        undefined,
                        undefined,
                        undefined,
                        {},
                        {},
                        {
                          "flexDirection": "row",
                          "justifyContent": "space-between",
                        },
                        {
                          "alignItems": "center",
                          "borderBottomWidth": 0.5,
                          "borderColor": "#cbcdd1",
                          "paddingBottom": 16,
                        },
                      ]
                    }
                  >
                    <Text
                      style={
                        [
                          {
                            "backgroundColor": "transparent",
                            "color": "#20303C",
                            "writingDirection": "ltr",
                          },
                          undefined,
                          undefined,
                          undefined,
                          undefined,
                          undefined,
                          undefined,
                          undefined,
                          undefined,
                          {
                            "color": "#f0f6fb",
                            "fontFamily": "Kanit-Medium",
                            "fontSize": 20,
                          },
                        ]
                      }
                    >
                      Ship Information
                    </Text>
                  </View>
                  <View
                    row={true}
                    style={
                      [
                        undefined,
                        undefined,
                        undefined,
                        undefined,
                        {},
                        {},
                        {
                          "flexDirection": "row",
                        },
                        {
                          "alignItems": "flex-start",
                          "borderBottomWidth": 0.5,
                          "borderColor": "#cbcdd1",
                          "gap": 16,
                          "paddingVertical": 10,
                        },
                      ]
                    }
                  >
                    <Image
                      accessibilityRole="image"
                      accessible={false}
                      assetGroup="icons"
                      onError={[Function]}
                      source={
                        {
                          "uri": "https://picsum.photos/id/237/200/300",
                        }
                      }
                      style={
                        [
                          undefined,
                          undefined,
                          undefined,
                          undefined,
                          undefined,
                          undefined,
                          undefined,
                          undefined,
                          false,
                          {
                            "height": 80,
                            "width": 100,
                          },
                          false,
                        ]
                      }
                    />
                    <View
                      style={
                        [
                          undefined,
                          undefined,
                          {
                            "flex": 1,
                          },
                          undefined,
                          {},
                          {},
                          {},
                          undefined,
                        ]
                      }
                    >
                      <Text
                        ellipsizeMode="tail"
                        numberOfLines={1}
                        style={
                          [
                            {
                              "backgroundColor": "transparent",
                              "color": "#20303C",
                              "writingDirection": "ltr",
                            },
                            undefined,
                            undefined,
                            undefined,
                            undefined,
                            undefined,
                            undefined,
                            undefined,
                            undefined,
                            {
                              "color": "#f0f6fb",
                              "fontFamily": "Kanit-Medium",
                              "fontSize": 20,
                            },
                          ]
                        }
                      />
                      <View
                        row={true}
                        style={
                          [
                            undefined,
                            undefined,
                            undefined,
                            undefined,
                            {},
                            {},
                            {
                              "flexDirection": "row",
                            },
                            {
                              "gap": 8,
                            },
                          ]
                        }
                      >
                        <Text
                          style={
                            [
                              {
                                "backgroundColor": "transparent",
                                "color": "#20303C",
                                "writingDirection": "ltr",
                              },
                              undefined,
                              undefined,
                              undefined,
                              undefined,
                              undefined,
                              undefined,
                              undefined,
                              undefined,
                              {
                                "color": "#9cd2ff",
                                "fontFamily": "Kanit-Regular",
                                "fontSize": 16,
                              },
                            ]
                          }
                        >
                          ShipInformationCard.textType
                        </Text>
                        <Text
                          style={
                            [
                              {
                                "backgroundColor": "transparent",
                                "color": "#20303C",
                                "writingDirection": "ltr",
                              },
                              undefined,
                              undefined,
                              undefined,
                              undefined,
                              undefined,
                              undefined,
                              undefined,
                              undefined,
                              {
                                "color": "#9cd2ff",
                                "fontFamily": "Kanit-Regular",
                                "fontSize": 16,
                              },
                            ]
                          }
                        />
                      </View>
                      <View
                        row={true}
                        style={
                          [
                            undefined,
                            undefined,
                            undefined,
                            undefined,
                            {},
                            {},
                            {
                              "flexDirection": "row",
                            },
                            {
                              "gap": 8,
                            },
                          ]
                        }
                      >
                        <Text
                          style={
                            [
                              {
                                "backgroundColor": "transparent",
                                "color": "#20303C",
                                "writingDirection": "ltr",
                              },
                              undefined,
                              undefined,
                              undefined,
                              undefined,
                              undefined,
                              undefined,
                              undefined,
                              undefined,
                              {
                                "color": "#9cd2ff",
                                "fontFamily": "Kanit-Regular",
                                "fontSize": 16,
                              },
                            ]
                          }
                        >
                          ShipInformationCard.textSize
                        </Text>
                        <Text
                          style={
                            [
                              {
                                "backgroundColor": "transparent",
                                "color": "#20303C",
                                "writingDirection": "ltr",
                              },
                              undefined,
                              undefined,
                              undefined,
                              undefined,
                              undefined,
                              undefined,
                              undefined,
                              undefined,
                              {
                                "color": "#9cd2ff",
                                "fontFamily": "Kanit-Regular",
                                "fontSize": 14,
                              },
                            ]
                          }
                        >
                          undefined m x undefined m x undefined m
                        </Text>
                      </View>
                    </View>
                  </View>
                  <View
                    style={
                      [
                        undefined,
                        undefined,
                        undefined,
                        undefined,
                        undefined,
                        undefined,
                        undefined,
                        undefined,
                      ]
                    }
                  />
                </View>
                <RCTScrollView
                  style={
                    {
                      "height": 600,
                    }
                  }
                >
                  <View>
                    <View
                      style={
                        [
                          undefined,
                          undefined,
                          undefined,
                          undefined,
                          undefined,
                          undefined,
                          undefined,
                          {
                            "backgroundColor": "#dbdee0",
                            "gap": 10,
                            "paddingHorizontal": 20,
                            "paddingTop": 20,
                          },
                        ]
                      }
                    >
                      <View
                        row={true}
                        spread={true}
                        style={
                          [
                            undefined,
                            undefined,
                            undefined,
                            undefined,
                            {},
                            {},
                            {
                              "flexDirection": "row",
                              "justifyContent": "space-between",
                            },
                            undefined,
                          ]
                        }
                      >
                        <Text
                          numberOfLines={0}
                          style={
                            {
                              "borderBottomWidth": 0,
                              "borderColor": undefined,
                              "color": "#828b92",
                              "fontFamily": "Kanit-Medium",
                              "fontSize": 16,
                              "lineHeight": undefined,
                              "textAlign": undefined,
                            }
                          }
                        >
                          TransactionProgress.textRentalID
                        </Text>
                        <Text
                          numberOfLines={0}
                          style={
                            {
                              "borderBottomWidth": 0,
                              "borderColor": undefined,
                              "color": "#4d5963",
                              "fontFamily": "Kanit-Regular",
                              "fontSize": 14,
                              "lineHeight": undefined,
                              "textAlign": undefined,
                            }
                          }
                        >
                          rentalId
                        </Text>
                      </View>
                      <View
                        row={true}
                        spread={true}
                        style={
                          [
                            undefined,
                            undefined,
                            {
                              "flex": 1,
                            },
                            undefined,
                            {},
                            {},
                            {
                              "flexDirection": "row",
                              "justifyContent": "space-between",
                            },
                            undefined,
                          ]
                        }
                      >
                        <View
                          style={
                            [
                              undefined,
                              undefined,
                              undefined,
                              undefined,
                              undefined,
                              undefined,
                              undefined,
                              {
                                "height": "100%",
                                "left": -40,
                                "width": "20%",
                              },
                            ]
                          }
                        >
                          <View
                            style={
                              [
                                {
                                  "backgroundColor": "rgba(1,0,0,0)",
                                },
                                {
                                  "flex": 1,
                                  "flexDirection": "row",
                                },
                              ]
                            }
                          >
                            <View
                              onLayout={[Function]}
                              style={
                                [
                                  {
                                    "alignItems": "center",
                                    "backgroundColor": "rgba(1,0,0,0)",
                                    "flexDirection": "row",
                                    "justifyContent": "space-around",
                                  },
                                  {
                                    "flexDirection": "column",
                                    "width": 40,
                                  },
                                ]
                              }
                            >
                              <View
                                accessibilityState={
                                  {
                                    "busy": undefined,
                                    "checked": undefined,
                                    "disabled": undefined,
                                    "expanded": undefined,
                                    "selected": undefined,
                                  }
                                }
                                accessible={true}
                                focusable={true}
                                onClick={[Function]}
                                onResponderGrant={[Function]}
                                onResponderMove={[Function]}
                                onResponderRelease={[Function]}
                                onResponderTerminate={[Function]}
                                onResponderTerminationRequest={[Function]}
                                onStartShouldSetResponder={[Function]}
                                style={
                                  [
                                    {
                                      "alignItems": "center",
                                      "flex": 1,
                                      "flexDirection": "row",
                                      "justifyContent": "center",
                                    },
                                    {
                                      "flexDirection": "column",
                                    },
                                  ]
                                }
                              >
                                <View
                                  collapsable={false}
                                  style={
                                    {
                                      "alignItems": "center",
                                      "backgroundColor": "transparent",
                                      "borderColor": "transparent",
                                      "borderRadius": 15,
                                      "borderWidth": 5,
                                      "height": 30,
                                      "justifyContent": "center",
                                      "overflow": "hidden",
                                      "width": 30,
                                      "zIndex": 2,
                                    }
                                  }
                                >
                                  <Text
                                    style={
                                      {
                                        "color": "transparent",
                                        "fontSize": 15,
                                        "overflow": "hidden",
                                      }
                                    }
                                  >
                                    1
                                  </Text>
                                </View>
                              </View>
                            </View>
                          </View>
                        </View>
                        <View
                          style={
                            [
                              undefined,
                              undefined,
                              undefined,
                              undefined,
                              undefined,
                              undefined,
                              undefined,
                              {
                                "height": "100%",
                                "position": "absolute",
                                "right": 0,
                                "width": "80%",
                              },
                            ]
                          }
                        >
                          <View
                            style={
                              [
                                {
                                  "backgroundColor": "rgba(1,0,0,0)",
                                },
                                {
                                  "flex": 1,
                                  "flexDirection": "row",
                                },
                              ]
                            }
                          >
                            <View
                              onLayout={[Function]}
                              style={
                                [
                                  {
                                    "alignItems": "center",
                                    "backgroundColor": "rgba(1,0,0,0)",
                                    "flexDirection": "row",
                                    "justifyContent": "space-around",
                                  },
                                  {
                                    "flexDirection": "column",
                                    "width": 30,
                                  },
                                ]
                              }
                            >
                              <View
                                accessibilityState={
                                  {
                                    "busy": undefined,
                                    "checked": undefined,
                                    "disabled": undefined,
                                    "expanded": undefined,
                                    "selected": undefined,
                                  }
                                }
                                accessible={true}
                                focusable={true}
                                onClick={[Function]}
                                onResponderGrant={[Function]}
                                onResponderMove={[Function]}
                                onResponderRelease={[Function]}
                                onResponderTerminate={[Function]}
                                onResponderTerminationRequest={[Function]}
                                onStartShouldSetResponder={[Function]}
                                style={
                                  [
                                    {
                                      "alignItems": "center",
                                      "flex": 1,
                                      "flexDirection": "row",
                                      "justifyContent": "center",
                                    },
                                    {
                                      "flexDirection": "column",
                                    },
                                  ]
                                }
                              >
                                <View
                                  collapsable={false}
                                  style={
                                    {
                                      "alignItems": "center",
                                      "backgroundColor": "#4cd669",
                                      "borderColor": "#4cd669",
                                      "borderRadius": 12.5,
                                      "borderWidth": 2,
                                      "height": 25,
                                      "justifyContent": "center",
                                      "overflow": "hidden",
                                      "width": 25,
                                      "zIndex": 2,
                                    }
                                  }
                                >
                                  <RNSVGSvgView
                                    align="xMidYMid"
                                    bbHeight="24"
                                    bbWidth="24"
                                    fill="none"
                                    focusable={false}
                                    height={24}
                                    meetOrSlice={0}
                                    minX={0}
                                    minY={0}
                                    style={
                                      [
                                        {
                                          "backgroundColor": "transparent",
                                          "borderWidth": 0,
                                        },
                                        {
                                          "flex": 0,
                                          "height": 24,
                                          "width": 24,
                                        },
                                      ]
                                    }
                                    vbHeight={24}
                                    vbWidth={24}
                                    width={24}
                                    xml="<svg width="24" height="24" viewBox="0 0 24 24" fill="none" xmlns="http://www.w3.org/2000/svg">
    <g clip-path="url(#clip0_77_28301)">
    <path d="M11.99 2C6.47 2 2 6.48 2 12C2 17.52 6.47 22 11.99 22C17.52 22 22 17.52 22 12C22 6.48 17.52 2 11.99 2ZM12 20C7.58 20 4 16.42 4 12C4 7.58 7.58 4 12 4C16.42 4 20 7.58 20 12C20 16.42 16.42 20 12 20ZM11.78 7H11.72C11.32 7 11 7.32 11 7.72V12.44C11 12.79 11.18 13.12 11.49 13.3L15.64 15.79C15.98 15.99 16.42 15.89 16.62 15.55C16.83 15.21 16.72 14.76 16.37 14.56L12.5 12.26V7.72C12.5 7.32 12.18 7 11.78 7Z" fill="white"/>
    </g>
    <defs>
    <clipPath id="clip0_77_28301">
    <rect width="24" height="24" fill="white"/>
    </clipPath>
    </defs>
    </svg>
    "
                                    xmlns="http://www.w3.org/2000/svg"
                                  >
                                    <RNSVGGroup
                                      fill={null}
                                      propList={
                                        [
                                          "fill",
                                        ]
                                      }
                                    >
                                      <RNSVGGroup
                                        clipPath="clip0_77_28301"
                                        fill={
                                          {
                                            "payload": 4278190080,
                                            "type": 0,
                                          }
                                        }
                                      >
                                        <RNSVGPath
                                          d="M11.99 2C6.47 2 2 6.48 2 12C2 17.52 6.47 22 11.99 22C17.52 22 22 17.52 22 12C22 6.48 17.52 2 11.99 2ZM12 20C7.58 20 4 16.42 4 12C4 7.58 7.58 4 12 4C16.42 4 20 7.58 20 12C20 16.42 16.42 20 12 20ZM11.78 7H11.72C11.32 7 11 7.32 11 7.72V12.44C11 12.79 11.18 13.12 11.49 13.3L15.64 15.79C15.98 15.99 16.42 15.89 16.62 15.55C16.83 15.21 16.72 14.76 16.37 14.56L12.5 12.26V7.72C12.5 7.32 12.18 7 11.78 7Z"
                                          fill={
                                            {
                                              "payload": 4294967295,
                                              "type": 0,
                                            }
                                          }
                                          propList={
                                            [
                                              "fill",
                                            ]
                                          }
                                        />
                                      </RNSVGGroup>
                                      <RNSVGDefs>
                                        <RNSVGClipPath
                                          fill={
                                            {
                                              "payload": 4278190080,
                                              "type": 0,
                                            }
                                          }
                                          name="clip0_77_28301"
                                        >
                                          <RNSVGRect
                                            fill={
                                              {
                                                "payload": 4294967295,
                                                "type": 0,
                                              }
                                            }
                                            height="24"
                                            propList={
                                              [
                                                "fill",
                                              ]
                                            }
                                            width="24"
                                            x="0"
                                            y="0"
                                          />
                                        </RNSVGClipPath>
                                      </RNSVGDefs>
                                    </RNSVGGroup>
                                  </RNSVGSvgView>
                                </View>
                              </View>
                            </View>
                          </View>
                        </View>
                      </View>
                    </View>
                  </View>
                </RCTScrollView>
              </View>
            </RCTScrollView>
          </View>
        </View>
      </View>
      <RNSScreenStackHeaderConfig
        backButtonInCustomView={false}
        backTitleVisible={true}
        backgroundColor="rgb(255, 255, 255)"
        color="rgb(0, 122, 255)"
        direction="ltr"
        disableBackButtonMenu={false}
        hidden={false}
        hideBackButton={false}
        largeTitleHideShadow={false}
        title="TrackTransactionsStat"
        titleColor="rgb(28, 28, 30)"
        topInsetEnabled={false}
        translucent={false}
      />
    </RNSScreen>
  </RNSScreenStack>
</RNCSafeAreaProvider>
`;
